### DISCUSSION TITLE
Question about database encryption

### DISCUSSION THREAD

**Discussion by makandre** [orientechnologies/orientdb#9893] [Q&A] [answered]
In the documentation located at https://orientdb.org/docs/3.2.x/security/Database-Encryption.html, there is the following special note:

> NOTE: Encryption parameters are not supported on remote protocol. It can be used only with in embedded or configuring the settings on the server.

I want to clarify what this really means.

Say I have an OrientDB that I start with `-Dstorage.encryptionKey="T1JJRU5UREJfSVNfQ09PTA=="`.  Databases created there "looks" encrypted (at least to my untrained eye poking around the wal files on disk).  Also subsequent restart of OrientDB without that key will fail (can't open the OSystem db).

Now from a remote Java client, it seems I can create a db with and without specifying the key--in both cases the db still "looks" encrypted, but more importantly, I can open and query the db without specifying the key.

Does what I'm describing make sense?  Is the database actually encrypted or not?  Is there a better way to determine that?

**Comment by tglman** [comment, marked as answer]
Hi,

In terms of security there are a few different features involved in your question that are supported by OrientDB.

First the one that you mention (https://orientdb.org/docs/3.2.x/security/Database-Encryption.html) is "Encryption At Rest" that means how data is obfuscated in the disk, this usually protect you from someone gaining not authorized access to the database files.

Then there is encryption on the network layer, this is probably the most important and is explained here: https://orientdb.com/docs/3.0.x/security/Using-SSL-with-OrientDB.html this make sure that plain data is not sent on the network.

Then there is authentication this is completely independent from the encryption, that can be done with basic user and password stored in the database or trough external authentication providers, here some details: https://orientdb.com/docs/3.0.x/security/Security-Config.html 

OrientDB as today do not have a way to provide database disk encryption key on creation/open of a database from remote, as well OrientDB do not provide a way to have encrypted data in memory that are possible to be read only by authorized.

Regards

**Reply by makandre** [reply]
Ok then, networking and authentication aside, I guess what I really want to know is whether the data is encrypted at rest when I start the OrientDB server with `storage.encryptionKey` provided.  Because it looks to me it is, but I want to confirm.

**Reply by tglman** [reply]
Yes,  with 3.2.x all the data is encrypted at rest when the encryption key is provided.

**Reply by ghost** [reply]
> OrientDB do not provide a way to have encrypted data in memory that are possible to be read only by authorized

If the data are not stored in memory, how can we query them since they are encrypted on the disk?
Hi, can you please explain this further? Can we query the encrypted data as if they are unencrypted, or not?
